FpsCheck.py

```diff
@@ -409,51 +409,47 @@ def find_invalid_alias_eSLDs(self, check_sets):
         """
-        for primary in check_sets:
-            curr_set = check_sets[primary]
-            if curr_set.ccTLDs:
-                for aliased_site in curr_set.ccTLDs:
-                    # first check if the aliased site is actually anywhere else
-                    # in the fps
-                    if aliased_site != primary:
-                        if curr_set.associated_sites:
-                            sites = curr_set.associated_sites 
-                        else:
-                            sites = []
-                        if curr_set.service_sites:
-                            sites += curr_set.service_sites
-                        if aliased_site not in sites:
-                            self.error_list.append(
-                                "The aliased site " + aliased_site + 
-                                " contained within the ccTLDs must be a " +
-                                "primary, associated site, or service site " +
-                                "within the firsty pary set for " + primary)
-                    # check the validity of the aliases
-                    aliased_domain, aliased_tld = aliased_site.split(".", 1)
-                    if aliased_tld in self.icanns:
-                        icann_check = self.icanns.union({"com"})
-                    else:
-                        icann_check = self.icanns
-                    plus1s = [(site, site.split(".")[0], site.split(".")[-1])
-                              for site in curr_set.ccTLDs[aliased_site]]
-                    for eSLD in plus1s:
-                        if eSLD[1] != aliased_domain:
-                            self.error_list.append(
-                                "The following top level domain must match: " 
-                                + aliased_site + ", but is instead: " 
-                                + eSLD[0])
-                        if eSLD[2] not in icann_check:
-                            self.error_list.append(
-                                "The provided country code: " + eSLD[2] + 
-                                ", in: " + eSLD[0] + 
-                                " is not a ICANN registered country code")
+        for primary, curr_set in check_sets.items():
+            if not curr_set.ccTLDs:
+                continue
+            for aliased_site in curr_set.ccTLDs:
+                # first check if the aliased site is actually anywhere else
+                # in the fps
+                if not curr_set.includes(aliased_site, False):
+                    self.error_list.append(
+                        "The aliased site " + aliased_site + 
+                        " contained within the ccTLDs must be a " +
+                        "primary, associated site, or service site " +
+                        "within the firsty pary set for " + primary)
+                # check the validity of the aliases
+                aliased_domain, aliased_etld = (aliased_site.split(".")[0],
```

Comment:
I wonder if there's a better word than `domain` we could use here, instead of `aliased_domain`? Strictly speaking, the whole string is the domain, including the TLD.

This code is really just trying to remove the eTLD from the domain. We called the remaining segment the "eSLD" in the submission guidelines, but I don't think that term is correct - the SLD of `example.co.uk` is `co.uk`, not `example` or `example.co`. Maybe the best choice is the most straightforward one: `aliased_site_without_etld` or similar?

I also think we should use the `publicsuffix2` library in order to identify the eTLD, instead of assuming that it's all the domain segments except the first one. But that can be done in a followup.

Reply:
Discussed this offline to resolve. Name changed, but it was decided `eSLD` was an appropriate term in general.